src/libs/elektra/key.c

```diff
@@ -403,6 +419,7 @@ int keyCopy (Key * dest, const Key * source)
 
 	// free old resources of destination
 	if (!test_bit (dest->flags, KEY_FLAG_MMAP_KEY)) elektraFree (destKey);
+	if (!test_bit (dest->flags, KEY_FLAG_MMAP_KEY)) elektraFree (destUKey);
```

Comment:
The `KEY_FLAG_MMAP_KEY` is a flag for the Key->key member. If I understand this correctly `ukey` is a member which is separately allocated. Then we need to add a separate flag for this field.

Reply:
I don't think we do. `key->key` and `key->ukey` are always changed at the same time. Either both are `NULL` or both are correctly allocated.

Reply:
In that case you're right, I was not aware of that.

Reply:
Such invariants should also be documented in the struct.